src/main/java/com/example/pokebattlez/battle/service/PokemonService.java

```diff
@@ -22,19 +25,53 @@ public PokemonService(RestTemplateBuilder restTemplateBuilder) {
         restTemplate = restTemplateBuilder.build();
     }
 
-    public void getBaseStats(Pokemon pokemon) {
+    public void getApiData(BattlePokemon battlePokemon) {
         ApiPokemon apiPokemon = restTemplate.getForObject(
-                String.format("https://pokeapi.co/api/v2/pokemon/%s", pokemon.getName()),
+                String.format("https://pokeapi.co/api/v2/pokemon/%s", battlePokemon.getName()),
                 ApiPokemon.class
         );
 
         assert apiPokemon != null;
```

Comment:
You want to throw a general assertion here? How about a custom exception at least?

Reply:
It's a bit of a patch job, gonna have to manage this way better since this should actually stop the entire battle.